wandb/sdk/interface/artifacts.py

```diff
@@ -985,13 +986,15 @@ def get_artifacts_cache() -> ArtifactsCache:
     global _artifacts_cache
     if _artifacts_cache is None:
         singleton = wandb.sdk.wandb_setup._WandbSetup._instance
-        cache_dir_base = (
-            wandb.run.settings.cache_dir
-            if wandb.run and wandb.run.settings.cache_dir
-            else singleton._settings.cache_dir
-            if singleton and singleton._settings.cache_dir
-            else env.get_cache_dir()
-        )
+        cache_dir_base = env.get_cache_dir()
+        if (
+            wandb.run
+            and not isinstance(wandb.run, RunDisabled)
+            and wandb.run.settings.cache_dir
+        ):
+            cache_dir_base = wandb.run.settings.cache_dir
+        elif singleton and singleton._settings.cache_dir:
+            cache_dir_base = singleton._settings.cache_dir
```

Comment:
Do we need `else env.get_cache_dir()` in case both `if` and `elif` conditions are false.

Reply:
No, because it's initialized to that value (L#989)